### PR TITLE
kvutils: Use consistent argument name casing in KVErrors

### PR DESCRIPTION
**Pull request by miklos-da**
Summary of changes:
* Use camelCased argument names consistent with the Scala style guide.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by nicu-da — COMMENTED** [review]
This all looks good to me. 
I haven't checked though how does this interacts with the reflection code that we have. Isn't it going to delete some of the current logic that we have? Or include the field twice? Old name and new name

**Comment by miklos-da** [comment]
> This all looks good to me. I haven't checked though how does this interacts with the reflection code that we have. Isn't it going to delete some of the current logic that we have? Or include the field twice? Old name and new name

It's going to override it, i.e., we won't use reflection to generate the default context.

**Review by stefanobaghino-da — APPROVED** [review]

**Review by nicu-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala [hunk 1]
```diff
@@ -62,13 +62,18 @@ object KVErrors extends LedgerApiErrorGroup {
...
+          minimumRecordTime: Instant,
+          maximumRecordTime: Instant,
       )(implicit loggingContext: ContextualizedErrorLogger)
           extends KVLoggingTransactionErrorImpl(
             cause =
-              s"Invalid ledger time: Record time is outside of valid range [$minimum_record_time, $maximum_record_time]"
-          )
+              s"Invalid ledger time: Record time is outside of valid range [$minimumRecordTime, $maximumRecordTime]"
+          ) {
+        override def context: Map[String, String] = Map(
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala:72]: It really concerns me that we could easily forget to override `context` for a particular error. I honestly think that having the logic in `BaseError` is dangerous.

To mitigate this, I suggest adding `override def context = kvContext` in `KvLoggingTransactionErrorImpl` and making `kvContext` an abstract method that _must_ be overridden by subclasses.

**Review by hubert-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala (hunk 1 above)
Comment (miklos-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala:68]: I share your concern, especially that the logic in `BaseError` is not transparent.
I have made changes similar to as you suggested, with the slight change that `KvLoggingTransactionErrorImpl` now explicitly overrides `context` with an empty `Map` so subclasses must override it to produce any error metadata.
Does this work for you?

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/errors/KVErrors.scala:68]: Yes, sounds good!

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**